libs/libm/newlib/Make.defs

```diff
@@ -18,21 +18,31 @@
 #
 ############################################################################
 
-NEWLIB_VERSION=4.3.0.20230120
+ifeq ($(CONFIG_LIBM_USE_NEWLIB_ESP32),y)
+  NEWLIB_VERSION=esp-4.1.0_20230425
+  NEWLIB_BASENAME=newlib-esp32
+  NEWLIB_URL_BASE=https://github.com/espressif/newlib-esp32/archive
+  NEWLIB_TARBALL=$(NEWLIB_VERSION).tar.gz
+else
+  NEWLIB_VERSION=4.3.0.20230120
```

Comment:
So 4.1.0 performs better than 4.3.0?

Reply:
4.1.0  is the latest version of newlib-esp32 I can find.
I guess 4.3.0 is too new to start adaptation in Espressif.

Reply:
> I mean, do we know what is the gap between 4.1.0 from `sourceware.org` vs espressif one?

The mainly changes in newlib-esp32 is for RISCV. The change for Xtensa is not so much. as far as I can find, sqrt() and fmod() are different. I also can see some changes in sourceware.org newlib v4.3.0 were not picked into current newllib-esp32. We hope Espressif can bump the source to the latest soon.